### PR TITLE
Update Keychron Q0

### PR DESCRIPTION
**Pull request by adophoxia**
## Description
Superseeds https://github.com/qmk/qmk_firmware/pull/21406, but this will only cover one board from each series update of PRs.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/q0/rev_0131/rules.mk [hunk 1]
```diff
@@ -1 +1,2 @@
-# This file intentionally left blank
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```
Comment (zvecr) [inline keyboards/keychron/q0/rev_0131/rules.mk:2]: This should not be re-introduced
```suggestion
# This file intentionally left blank
```

File: keyboards/keychron/q0/rev_0130/rules.mk [hunk 2]
```diff
@@ -1 +1,2 @@
-# This file intentionally left blank
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```
Comment (zvecr) [inline keyboards/keychron/q0/rev_0130/rules.mk:2]:
```suggestion
# This file intentionally left blank
```

File: keyboards/keychron/q0/info.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        "audio": false,
+        "backlight": false,
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true,
+        "rgblight": false
```
Comment (zvecr) [inline keyboards/keychron/q0/info.json:16]:
```suggestion
        "bootmagic": true,
        "command": false,
        "console": false,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "rgb_matrix": true
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by drashna** [comment]
Since this is moving the folders around, this needs an entry in https://github.com/qmk/qmk_firmware/blob/master/data/mappings/keyboard_aliases.hjson at the very least.

File: keyboards/keychron/q1v1/q1.c
Comment (drashna) [inline keyboards/keychron/q1v1/q1.c:1]: this file needs to be renamed to q1v1.c

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q1v1/q1.c
Comment (adophoxia) [reply keyboards/keychron/q1v1/q1.c:1]: I must've mistakenly pulled the `keychron-q1v1-update `branch for my `Update Keychron Q1` PR and merged it, seeing how even `q1/readme.md` was changed. I'll deal with this right now.

**Pull request closed without merging**